api/v1alpha1/hostedcluster_types.go

```diff
@@ -355,6 +362,10 @@ const (
 	// UnmanagedEtcdAvailable indicates whether a user-managed etcd cluster is
 	// healthy.
 	UnmanagedEtcdAvailable ConditionType = "UnmanagedEtcdAvailable"
+
+	// InvalidHostedClusterConfiguration indicates (if status is true) that the
+	// ClusterConfiguration specified for the HostedCluster is not valid.
+	InvalidHostedClusterConfiguration ConditionType = "InvalidConfiguration"
```

Comment:
in NodePool I'm using Valid"Property" conditions because the word nuance is aligned with the bool value and that results more intuitive to me e.g valid|true (good) valid|false (bad). Whereas here to me is slightly less intuitive e.g Invalid|true (good). Thoughts for one vs the other one?

Reply:
I have no preference. According to API conventions (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) it's whatever is easier for the user to understand. My thinking was that this would be a condition that would only show up if there was a problem with the config, otherwise it would not clutter status.
However, reading the conventions doc, it seems that it's better that the condition always be there (using Unknown if you don't know whether it's true or not). So I can change it to the positive.